### DISCUSSION TITLE
Public vs Internal for getOrCreateAnvilConfig?

### DISCUSSION THREAD

**Discussion by CPix18** [Cyfrin/foundry-full-course-cu#2681] [General]
Hello, I recently came across the video series for this after looking through Foundry docs and have been really enjoying the course. At this point I'm just about an hour in on the second video series and kept getting the red line under my getOrCreateAnvilConfig function in the constructor. This may be some error on vsCode bc I previously messed up the spelling (getOrCreatAnvilConfig) and it is suggesting I may need to use that even though there is no reference in my code with that misspelling anywhere. Plus my tests pass using forge test. However when I was troubleshooting, ChatGPT said something about since that function changes state and is in the constructor it should be labeled as internal instead of public. Here is the response from ChatGPT "The reason why getOrCreateAnvilConfig cannot be public is that it involves operations that aren't pure or view, and it relies on the vm context, which should only be used in specific scenarios. By marking it as internal, you ensure it's appropriate for use within the constructor." Basically I'm wondering if this is actually correct or if ChatGPT is just pretending to know this for a fact.

If anyone knows how to resolve my error on vsCode that would be awesome as well. Maybe something with the cache? Here is a screenshot.
<img width="568" alt="Screen Shot 2024-10-14 at 6 59 27 PM" src="https://github.com/user-attachments/assets/27808e77-6ce2-4871-a74a-f8c7b27a063a">

**Comment by EngrPips** [comment]
Sometimes `vscode` marks stuff with a red line, but the code is perfectly fine, and how you will know is that you can compile your code successfully and run tests successfully. ChatGPT may hallucinate, and that is why you are learning so that you don't blindly rely on it and know when it is correct and when it is spitting jargon.

**Reply by CPix18** [reply]
Thanks for the reply! Yes I was able to get it working by saving, closing down vscode and reopening. Appreciate the help!

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
